jobs/build.Jenkinsfile

```diff
@@ -345,24 +345,33 @@ lock(resource: "build-${params.STREAM}") {
             tar -cf - tmp/kola/ | xz -c9 > kola-run.tar.xz
             """)
             archiveArtifacts "kola-run.tar.xz"
-            if (!pipeutils.checkKolaSuccess("tmp/kola", currentBuild)) {
-                return
+            if (!pipeutils.checkKolaSuccess("tmp/kola")) {
+                error('Kola:QEMU')
             }
         }
 
         // Kola QEMU Upgrade tests
         parallelruns['Kola:QEMU Upgrade'] = {
+            // If upgrades are broken `cosa kola --upgrades` might
+            // fail to even find the previous image so we wrap this
+            // in a try/catch so ALLOW_KOLA_UPGRADE_FAILURE can work.
             try {
                 shwrap("""
                 cosa kola --rerun --upgrades --no-test-exit-error
                 tar -cf - tmp/kola-upgrade | xz -c9 > kola-run-upgrade.tar.xz
                 """)
                 archiveArtifacts "kola-run-upgrade.tar.xz"
-                if (!params.ALLOW_KOLA_UPGRADE_FAILURE && !pipeutils.checkKolaSuccess("tmp/kola-upgrade", currentBuild)) {
-                    return
+                if (!pipeutils.checkKolaSuccess("tmp/kola-upgrade")) {
+                    error('Kola:QEMU Upgrade')
                 }
             } catch(e) {
-                if (!params.ALLOW_KOLA_UPGRADE_FAILURE) {
+                if (params.ALLOW_KOLA_UPGRADE_FAILURE) {
+                    // reset result back to !FAILED and warn
+                    currentBuild.result = ''
```

Comment:
This isn't necessary. Throwing an exception doesn't change the result.

Reply:
I meant to drop that bit in the 2nd commit and forgot. Check now.